src/utils/utils.rs

```diff
@@ -587,6 +587,22 @@ where
     })
 }
 
+pub fn delete_dir_contents(dir_path: &Path) {
+    let dir_contents = fs::read_dir(&dir_path);
```

Comment:
I've just spotted that you have an unnecessary borrow here.  `dir_path` is already a `&Path` so you are creating a `&&Path` which the compiler is then automatically dereffing.

```suggestion
    let dir_contents = fs::read_dir(dir_path);
```